Implemented basic sections

Hysterelius opened the pull request:
On the way to fixing #143, have created the basic feature.

Here are the things lacking:
- [x] Custom CSS styling
- [x] Multi-lingual series
- [ ] Override pagination
- [ ] Collapsable for large series

I have just opened this so you can see my progress so far. You can see what I have done on the _Maths_ page.

Hysterelius commented:
I'm pretty much finished on this PR and feature, let me know what you think of the styling. 
Also, I decided not to override pagination as it could become too confusing, but it could be changed. 

I am not sure how collapsibility would be implemented, but it could always be added later. 

It would be great if you could have a look over my changes.

Jieiku commented:
Need to also add the option to the overrides scss file: https://github.com/Jieiku/abridge/blob/c44af9da6d9db0f420eabce146058ae0ac48b353/COPY-TO-ROOT-SASS/abridge.scss#L185

I like the placement of the series block, it looks good, but I would get rid of the parenthesis and the count from the title so instead of:

```
Features (10 part
Series)
```

I would simply have :

`Features Series`

The reader can infer how many parts there are from the numbered list, this should keep this header to a single line in most cases and make the appearance better.

![2023-09-17_11-06-49](https://github.com/Jieiku/abridge/assets/106644/0f250343-739a-4410-9fd6-617283e68126)

Need to get the margin spacing on the Series Block and the posts table of content to look the same so that it all looks consistent:

![2023-09-17_11-11-55](https://github.com/Jieiku/abridge/assets/106644/4ef1830d-b97d-48ec-b7cc-dc6892f92e3f)

I recently removed some padding above the word "Index" however I may have gone a little too far, the idea here is that on a very large pages with a loooooong table of contents, to try and make as efficient use of the page space as possible.

The easiest way to accomplish this is to change this
`<h4><a href="{{ tag.permalink | safe }}{%- if uglyurls %}index.html{%- endif %}">{{ tag.name }} ({{ parts_string }})</a></h4>`

To this:

<a class="si" href="{{ tag.permalink | safe }}{%- if uglyurls %}index.html{%- endif %}">{{ tag.name }} ({{ parts_string }})</a>`

Then if you think there needs to be a little more spacing you can adjust the si class.

PS: The same treatment needs to be done to the recent posts block (keeping a consistent look with spacing), however there is other work that is also on my todo list for the recent posts block. When I initially implemented the feature I reused some code and never went back and cleaned that up. I would prefer that people could use just table of contents without also having to use the recent posts block. In config.toml I have recent = true, that value should only affect the recent posts block, then I should also have a toc = true, so that they can use only one of the features if they wanted.

https://github.com/Jieiku/abridge/blob/c44af9da6d9db0f420eabce146058ae0ac48b353/config.toml#L114

So for the recent posts block you could clean that up too to match the series block and table of contents block as far as spacing goes if you wanted. It also needs its logic separated but I can do that later if your not wanting to mess with it.


Also I see you reused some of the scss from the toc for the series-toc which is good, it keeps it consistent, because series-toc is not a sticky element you can remove the top attribute:

![2023-09-17_11-41-03](https://github.com/Jieiku/abridge/assets/106644/549999bb-14a4-4e4f-8e7e-ef3fa5e0017f)

I noticed you made use of the cursor attribute in the scss, `a` links should have that automatically, is there a way those could be removed so that we can keep the css as light as possible.

There might also be other scss elements that can be removed from this pull request, but I am not certain without testing removing these one a time and looking at the result, you could see if there is any extra scss that can be removed. You don't have to go out of your way to make it lighter, like restructuring the heck out of it, but if there are any attributes that are simply not doing anything then those should be removed.

(I recently did a LOT of cleanup for the main Navigation Menu, but have not pushed the commit for it yet, and will probably wait to push my commit until we get yours in so that you do not have to refactor)

Jieiku commented:
## Override pagination
I think it is a good idea to override the pagination, but it may not even be necessary if the posts are within their own section. The simplest solution is to make sure that series are [true sections](https://www.getzola.org/documentation/content/section/), and by that I mean any articles that are part of a series would go into their own section folder:

![2023-09-17_14-08-08](https://github.com/Jieiku/abridge/assets/106644/0ca00b98-e4f6-4f5e-9882-1705c6ec81a8)

The _index.md file would simply have the metadata and any references to used templates, the rest of the md files in the folder would be the content that is part of that series, this would also keep the content organized on disk. (using a transparent section would probably work best here so that the pagination stays true on the index.)

As a benefit/side affect the result in the URL would be that instead of this:

https://abridge.netlify.app/overview-embed-youtube/
https://abridge.netlify.app/overview-embed-vimeo/
https://abridge.netlify.app/overview-embed-streamable/

you would have:

https://abridge.netlify.app/overview-video-sites/overview-embed-youtube/
https://abridge.netlify.app/overview-video-sites/overview-embed-vimeo/
https://abridge.netlify.app/overview-video-sites/overview-embed-streamable/

Another benefit to this is that on the `_index.md` you could also specify an overview page.

so maybe you have a page that simply discusses whats in that section, and if this is not desired you can instead set render=false, which is what I do for pages: 

https://github.com/Jieiku/abridge/blob/c44af9da6d9db0f420eabce146058ae0ac48b353/content/pages/_index.md?plain=1#L2

## Collapsable for large series
For collapsible menu on the series.... you could probably create it similar to a paginator, and have a configurable max length, so any series up to 20 items could display the full list, and any series beyond 20 items could be paginated, eg:

1. article one
...
5. article five
6. article six
7.  article seven
8.  article eight
9.  article nine
...
32. article 32

Hysterelius commented:
Thank you for all your advice and feedback, especially when you're busy!

--------

> `Features (10 part Series)`
> I would simply have:
> `Series of 10`

I believe that 10 part Series is a better choice as it simplifies translations and, in my opinion, is more intuitive.

> So for the recent posts block you could clean that up too to match the series block and table of contents block as far as spacing goes if you wanted. It also needs its logic separated but I can do that later if your not wanting to mess with it.

Yep, I'll have a look at making a consistent style for these _blocks_.

> I think it is a good idea to override the pagination, but it may not even be necessary if the posts are within their own section. The simplest solution is to make sure that series are [true sections](https://www.getzola.org/documentation/content/section/),

I'll see what changes that `true sections` need and whether they will work.

> For collapsible menu on the series.... you could probably create it similar to a paginator, and have a configurable max length, so any series up to 20 items could display the full list, and any series beyond 20 items could be paginated, so for instance if you were on article 7 of 32:

I'll have a play with this and see how it goes.

> Honestly they can both be saved for later if you would rather me work on them later. (that may actually be best because I have some commits for https://github.com/Jieiku/abridge/issues/129 that would probably be useful here.)

I'll still just have a little look how I go, and we can see from there.

-------

### Summary (for me):
- [ ] Cut title
- [ ] Clean up spacing on _blocks_
- [ ] Remove unused css: `cursor`, `top`
- [ ] True sections for pages: fix tags
- [ ] WIP: collapsibility

Hysterelius commented on templates/page.html:
```diff
@@ -45,6 +45,35 @@
+
+
+    <div class="series-mobile">
+      <h4><a href="{{ tag.permalink | safe }}{%- if uglyurls %}index.html{%- endif %}"></a>{{ tag.name }} ({{ parts_string }})</a></h4>
+      <ol>
+      {%- for inner in tag.pages | sort(attribute="year") %}
+      {%- if inner.content == page.content %}
+      <li><a href="{{ inner.permalink | safe }}{%- if uglyurls %}index.html{%- endif %}"><b>{{ inner.title }}</b></a></li>
+      {% else %}
+      <li><a href="{{ inner.permalink | safe }}{%- if uglyurls %}index.html{%- endif %}">{{ inner.title }}</a></li>
+      {%- endif %}
+      {%- endfor %}
+      </ol>
```
Is there a way to make this into a block so the code doesn't have to repeated below?

Jieiku replied:
I wouldn't worry about that for now, its ok for some duplicated code, especially in the templates since once the site is generated it wont affect performance.

Hysterelius commented on templates/page.html:
```diff
@@ -67,7 +96,23 @@
     <div class="toc">
+      {%- if page.extra.section %}
+      <div class="series-toc">
+        <h4><a href="{{ tag.permalink | safe }}{%- if uglyurls %}index.html{%- endif %}">{{ tag.name }} ({{ parts_string }})</a></h4>
+        <ol>
+        {%- for inner in tag.pages | sort(attribute="year") %}
+        {%- if inner.content == page.content %}
+        <li><a href="{{ inner.permalink | safe }}{%- if uglyurls %}index.html{%- endif %}"><b>{{ inner.title }}</b></a></li>
+        {% else %}
+        <li><a href="{{ inner.permalink | safe }}{%- if uglyurls %}index.html{%- endif %}">{{ inner.title }}</a></li>
+        {%- endif %}
+        {%- endfor %}
+        </ol>
```
On these lines below

Hysterelius replied:
You'll have to block everything below the `<h4>`, to allow the title to be changed varying on whether it is toc or _in place_.

Hysterelius commented:
> So for the recent posts block you could clean that up too to match the series block and table of contents block as far as spacing goes if you wanted. It also needs its logic separated but I can do that later if your not wanting to mess with it.

I have fixed this just by putting this at the bottom of `abridge.scss` file:
```
.toc h1 {
  margin: 0;
}
```
It could probably be under its own `@if` block, but I don't really see the point if it is just one element.

Jieiku commented:
Posts from a transparent section should appear in recent posts block, but your correct that a non transparent section likely would not with the current recent posts block. Give it a try with `transparent = true` in the `_index.md` for a series section.

Jieiku commented:
I pushed a small commit, is this ready to be merged? I just took a look and it looks good to me.

Hysterelius commented:
The basic functionality is ready for deployment. I am currently working on a MVP for condensing of series and will have a look at folder sections.

Series are currently based on a shared tag, but it may require a breaking change if we were to switch over to _true sections_.
<details>
<summary>What CSS I removed:</summary>
I have cut down on the css as much as I could:

- Added new `@if` for series
- Created new `series` parent class to house shared styling
- Removed `min-width` on in-place blocks
- Removed unnecessary `text-decoration: none`
<\details>

Jieiku commented:
Thanks! Looks good, going to merge now.

Hysterelius commented:
> Honestly they can both be saved for later if you would rather me work on them later. (that may actually be best because I have some commits for https://github.com/Jieiku/abridge/issues/129 that would probably be useful 

I have started to work a bit on the logic for the pages and would like to know if it would be best time to start to implement it?

Also, have you made in progress on true sections and pagination? 
Otherwise, I am happy to have a lock at some more logic to deal with pagination.

Jieiku commented:
I mostly just restructured the css for the side column blocks so that the classes can be reused. I also go all of my other pending commits in.

I tested creating a section just now, and using transparent=true, and the pagination at the bottom of the page still went by date instead of section. I used transparent=true so that the posts would show up in the recent posts block...

I have multiple ideas for solving this, but all of them involve a good deal of work. As far as Organizing on disk goes transparent section work without issue, I just pushed a commit here: 

https://github.com/Jieiku/abridge/commit/ba30a59c478715056ac2d01d16aac75a22ea0628

Now you will see it them under the video-streaming-sites transparent section:

https://abridge.netlify.app/video-streaming-sites/overview-embed-youtube/

and if you go here:

https://abridge.netlify.app/video-streaming-sites/

you will see a 404 page, but if you change render = true for that section, then you will see the page that describes that section: https://github.com/Jieiku/abridge/blob/ba30a59c478715056ac2d01d16aac75a22ea0628/content/video-streaming-sites/_index.md?plain=1#L2 (you can test it with zola serve)

To NOT use a transparent section would mean that multiple parts of Abridge would have to account for other sections where that is desired... the index that lists the posts, the recent posts block, even the archives page. So I am not sure if it is a good thing to pursue. (does it solve any real problems?)

It may be simpler to use transparent section for organization on disk, and for the pagination at the bottom of the page we could just rework it so that it stays within that series...

I do not have time for either solution at the moment.
